block-producer/src/store/mod.rs

```diff
@@ -181,9 +181,15 @@ impl Store for DefaultStore {
             nullifiers.into_iter().collect()
         };
 
-        let tx_inputs = TxInputs {
-            account_hash,
-            nullifiers,
+        let tx_inputs = match account_hash {
+            Some(hash) if hash == Digest::default() => TxInputs {
+                account_hash: None,
+                nullifiers,
+            },
+            _ => TxInputs {
+                account_hash,
+                nullifiers,
+            },
```

Comment:
Under what conditions would it be valid for account hash to come back as `None` from the store? Basically, if have cases here:

1. `Some(account_hash)` and `account_hash != Digest::default()` - this means the account is in the store.
2. `Some(account_hash)` and `account_hash == Digest::default()` - this means the account is new.
3. `None` - shouldn't this be an error?

Reply:
Indeed it should be an error, the store should not return a None value.

Reply:
This new change should fix it.